src/datasets/io/json.py

```diff
@@ -81,9 +82,13 @@ def write(self) -> int:
         _ = self.to_json_kwargs.pop("path_or_buf", None)
         orient = self.to_json_kwargs.pop("orient", "records")
         lines = self.to_json_kwargs.pop("lines", True)
+        compression = self.to_json_kwargs.pop("compression", None)
+
+        if compression not in [None, "gzip"]:
+            raise NotImplementedError(f"Datasets currently does not support {compression} compression")
 
         if isinstance(self.path_or_buf, (str, bytes, os.PathLike)):
-            with open(self.path_or_buf, "wb+") as buffer:
+            with IOHandler(self.path_or_buf, "wb", compression) as buffer:
```

Comment:
Any reason for removing the updating mode  "+"?

Reply:
afair `bz2` and `gzip` don't have `wb+` mode. I didn't see any mode ending with + in these compression formats in python docs